apps/openmw/mwphysics/actor.cpp

```diff
@@ -150,6 +150,16 @@ void Actor::setCanWaterWalk(bool waterWalk)
     {
         mCanWaterWalk = waterWalk;
         updateCollisionMask();
+	if (waterWalk) {
+	    ESM::Position pos = mPtr.getRefData().getPosition();
+	    if (pos.pos[2] < -100) {
```

Comment:
Don't use the number -100. That is wrong in several ways.

One, you can not assume the water to be at z=0. You need to take the cells' water level into account.

Second, you are assuming the player/actor is 100 units high. Use the actual height of the actor instead.